libtiny_tui/src/lib.rs

```diff
@@ -65,6 +64,28 @@ impl TUI {
 
         (TUI { inner }, rcv_ev)
     }
+
+    /// Create a test instance that doesn't render to the terminal, just updates the termbox
+    /// buffer. Useful for testing. See also [`get_front_buffer`](TUI::get_front_buffer).
+    pub fn run_test(width: u16, height: u16) -> (TUI, mpsc::Receiver<Event>) {
+        let tui = Rc::new(RefCell::new(tui::TUI::new_test(width, height)));
+        let inner = Rc::downgrade(&tui);
+
+        let (snd_ev, rcv_ev) = mpsc::channel(10);
+
+        // We don't need to handle SIGWINCH in testing so the receiver end is not used
+        let (snd_abort, _rcv_abort) = mpsc::channel::<()>(1);
+
+        // Spawn input handler task
+        spawn_local(input_handler(tui, snd_ev, snd_abort));
```

Comment:
This won't work, we have to create a `Stream` for input events (probably a channel so that we can return the sender to the caller and write to it in tests) and pass it to `input_handler`.